### PR TITLE
327 & 328 organisation maintainer rls

### LINKED ISSUE #327: Organisation mantainer should be able to see all (linked) software
## Organisation maintainer should see all software 

The organisation maintainer should be able to see all software that mentions the organisation independent from the fact if the software is published or not and if organisation maintainer is the maintainer of the software.

## The software counts shown in the menu item
This count represents all software that links to the organisation. This includes: published, not published and denied/blocked software items.

### LINKED ISSUE #328: Organisation maintainer should be able to pin projects and deny affiliation
Similair to software section on the organisation page, the organisation maintainer should be able to:

* pin project at organisation page: pinned projects are shown first
* deny affiliation with the project: this will hide organisation on the project page

### PR DESCRIPTION
**Pull request by dmijatovic**
# Organisation maintainer features for software and projects

Closes #327 
Closes #328
Related to #346
Related to #350 

Changes proposed in this pull request:
* Organisation maintainer is able to see all software affiliated with the organisation incl. not yet published software items
* Organisation maintainer is able to pin/unpin, deny/accept project affiliation with organisation, and see not yet published projects linking to organisation

How to test:
* `docker-compose down --volumes`: to remove old
* `docker-compose build`: to build all services
* `docker-compose up`: to start
* perform data migration
* start app and login as test user
* make yourself (primary) maintainer of an organisation (or all organisations)
* navigate to your organisation
* pin/unpin software, reload page to see impact, you can also logout and view organisation page as "viewer"
* login as different user
* create new software, add organisation of the "previous" maintainer
* navigate to organisation, you should not be able to see new software you just created because it is not published yet
* logout and login as "previous" user
* navigate to your organisation, you should be able to see not published software from the "second" user in the list
* create new project,
* navigate to your organisation page, you should be able to see new project, even if not published yet
* deny/approve affiliation from the organisation page
* pin/unpin project from organisation page

## Example features organisation maintainer on project page
image

PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [x] Link to a GitHub issue
*   [ ] Update documentation
*   [ ] Tests

### CODE REVIEW

**Comment by ruester** [comment]
> make yourself (primary) maintainer of an organisation (or all organisations)

This is done by manipulating the entries in the DB, right? Will there be an easier way in the future? For example having an administration login who is allowed to change everything (like the local authentication but with a configured password and a pre-defined user)?

**Comment by ruester** [comment]
I've gone through the testing steps and this is what I noticed:
- I am still able to publish software as the second user, but I am not able to publish it as the first user (maintainer of the organisation) -> I think this is not part of this PR and is also intended for now?
- being the second user I see my created software entry not with the eye-icon as the first user would see, I would expect that the eye-icon is there regardless of being a maintainer of the organisation or not, and this could also be the case for the "My software" view (for the project view this is exactly as expected -> if not published: show eye-icon)
- if a project is denied from the organisation the second user is not anymore able to remove the organisation from the project (what if it was filled in by mistake?), same for software
- the denied affiliations are counted in the grid and therefore differ if I am logged in or not (this may only be a problem if I deny 10 project affiliations and then maybe wonder why there are so many projects associated to my organisation.. but the "public view", not being logged in, still shows it right)
- but the software count is wrong for denied affiliations:

image

vs.

image
(not being logged in)

or:

image
(count 4 on the left, but only 3 software entries on the right)

**Comment by ruester** [comment]
I also did not see this pin icon which is shown in the PR description. For me the software entry only gets blue if I pin it.

**Comment by dmijatovic** [comment]
@ruester Thanks for detailed review. Yes, we do plan to build admin section for assigning the maintainers. I'll come back concerning the other points.

**Comment by cmeessen** [comment]
A user that added an organisation to a software that denied the affiliation cannot remove the organisation again. This is the view in Edit Software -> Organisations, after affiliation was denied:

image

**Comment by ruester** [comment]
- I cannot see a denied affiliation as the user who created the affiliation, only the organisation maintainer sees the "forbidden sign", but it may be intentional for now
- Point 3: understood now, thanks. This prevents the user from creating the affiliation over and over again and the maintainer would have no control over the affiliations :+1:  

Otherwise everything works as expected now (or will be handled in new issues) :+1:

**Review by ruester — APPROVED** [review]

File: frontend/components/organisation/units/index.tsx [hunk 1]
```diff
@@ -127,6 +128,19 @@ export default function ResearchUnits({organisation, session, isMaintainer}:
...
           addUnitToCollection(newUnit)
+          if (newUnit.id && session.user?.account) {
+            // add maintainer to org
+            const resp = await addMaintainerToOrganisation({
+              organisation: newUnit.id,
+              account: session.user?.account,
+              token: session.token,
+              frontend: true
+            })
+            debugger
```
Comment (ruester) [inline frontend/components/organisation/units/index.tsx:139]: could be switched to a comment

Comment (dmijatovic) [reply frontend/components/organisation/units/index.tsx:139]: @ruester Well spotted, I fixed it in #346 which will be merged into this branch and then to main

**Pull request merged**